### PR TITLE
Feature for Plex binding

### PR DESCRIPTION
**Pull request by idserda**
This PR adds a feature for the Plex binding and makes it configurable in the Paper UI.

### CODE REVIEW

File: bundles/binding/org.openhab.binding.plex/ESH-INF/binding/binding.xml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    <description>The Plex Binding connects to a Plex Media Server to track the state of multiple clients.</description>
+    <author>Jeroen Idserda</author>
+
+    <service-id>org.openhab.plex</service-id>
+
+    <config-description>
+        <parameter name="host" type="text" required="true">
+            <context>network-address</context>
+            <label>Host</label>
+            <description>IP address of the Plex server</description>
```
Comment (watou) [inline bundles/binding/org.openhab.binding.plex/ESH-INF/binding/binding.xml]: Could this be changed to support hostnames like your recent change?  (And that would mean the `<context>` element would no longer apply.)

File: bundles/binding/org.openhab.binding.plex/ESH-INF/binding/binding.xml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        </parameter>
+        <parameter name="port" type="text">
```
Comment (watou) [inline bundles/binding/org.openhab.binding.plex/ESH-INF/binding/binding.xml]: `type="integer"`

File: bundles/binding/org.openhab.binding.plex/ESH-INF/binding/binding.xml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        </parameter>
+        <parameter name="refresh" type="integer">
+            <label>Refresh</label>
+            <description>Refresh interval in ms, only used for polling online/offline status of clients.</description>
+            <default>5000</default>
+            <unitLabel>ms</unitLabel>
+        </parameter>
+        <parameter name="username" type="text">
+            <label>Username</label>
+            <description>If you're using Plex Home you need to supply the username and password of your Plex account here. If you don't want to enter your credentials you can also directly set your account token below instead.</description>
```
Comment (watou) [inline bundles/binding/org.openhab.binding.plex/ESH-INF/binding/binding.xml]: Possible to shorten the description a bit?  I think it won't format well in the UI.

File: features/openhab-addons-external/src/main/resources/conf/plex.cfg [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# IP address of the Plex server
```
Comment (watou) [inline features/openhab-addons-external/src/main/resources/conf/plex.cfg]: or hostname?

**Comment by watou** [comment]
With the support to configure the binding from the Paper UI, we enter a mode where some configuration elements could be other basic types like Integer and Boolean.  Therefore, all the constructs [here](https://github.com/openhab/openhab/blob/master/bundles/binding/org.openhab.binding.plex/src/main/java/org/openhab/binding/plex/internal/PlexBinding.java#L167) like 

```
(String) configuration.get("X");
```
should be changed to
```
Objects.toString(configuration.get("X"), null);
```
In this way, the configuration dictionary can contain values of any type and the binding remains unconcerned.

**Comment by idserda** [comment]
Thanks for you feedback! Made the requested changes.

File: bundles/binding/org.openhab.binding.plex/ESH-INF/binding/binding.xml (hunk 3 above)
Comment (watou) [inline bundles/binding/org.openhab.binding.plex/ESH-INF/binding/binding.xml]: Better to have more description than not, so that's good!

**Comment by watou** [comment]
Thank you, @idserda!  If [my comment](https://community.openhab.org/t/openhab2-plex/14036/62?u=watou) equates to nothing of concern, then this PR LGTM.

**Pull request merged**